src/test/java/fi/helsinki/cs/tmc/core/communication/ExerciseSubmitterTest.java

```diff
@@ -127,132 +128,126 @@ public void testFindCourseByCorrectPath() throws IOException, TmcCoreException {
-        this.courseSubmitter =
-                new ExerciseSubmitter(rootFinder, langs, urlCommunicator, tmcApi, settings);
-        rootFinder.setReturnValue(testPath);
-        String pastePath = "https://tmc.mooc.fi/staging/paste/ynpw7_mZZGk3a9PPrMWOOQ";
-        String result = courseSubmitter.submitPaste(testPath);
+            TmcCoreException, URISyntaxException, NoLanguagePluginFoundException {
+        Path testPath
+                = Paths.get(
+                        "home",
+                        "test",
+                        "2014-mooc-no-deadline",
+                        "viikko1",
+                        "viikko1-Viikko1_001.Nimi");
+        settings.setCurrentCourse(rootFinder.getCurrentCourse(testPath.toString()).or(new Course()));
+        this.courseSubmitter
+                = new ExerciseSubmitter(rootFinder, langs, urlCommunicator, tmcApi, settings);
+        rootFinder.setReturnValue(testPath.toString());
+        URI pastePath = URI.create("https://tmc.mooc.fi/staging/paste/ynpw7_mZZGk3a9PPrMWOOQ");
+        URI result = courseSubmitter.submitPaste(testPath);
         assertEquals(pastePath, result);
     }
 
     @Test
     public void submitWithPasteAndCommentReturnsPasteUrl()
             throws IOException, ParseException, ExpiredException, IllegalArgumentException,
-                    TmcCoreException, URISyntaxException, NoLanguagePluginFoundException {
-        String testPath =
-                Paths.get(
-                                "home",
-                                "test",
-                                "2014-mooc-no-deadline",
-                                "viikko1",
-                                "viikko1-Viikko1_001.Nimi")
-                        .toString();
-        settings.setCurrentCourse(rootFinder.getCurrentCourse(testPath).or(new Course()));
-        this.courseSubmitter =
-                new ExerciseSubmitter(rootFinder, langs, urlCommunicator, tmcApi, settings);
-        rootFinder.setReturnValue(testPath);
-        String pastePath = "https://tmc.mooc.fi/staging/paste/ynpw7_mZZGk3a9PPrMWOOQ";
-        String result = courseSubmitter.submitPasteWithComment(testPath, "Commentti");
+            TmcCoreException, URISyntaxException, NoLanguagePluginFoundException {
+        Path testPath
+                = Paths.get(
+                        "home",
+                        "test",
+                        "2014-mooc-no-deadline",
+                        "viikko1",
+                        "viikko1-Viikko1_001.Nimi");
+        settings.setCurrentCourse(rootFinder.getCurrentCourse(testPath.toString()).or(new Course()));
+        this.courseSubmitter
+                = new ExerciseSubmitter(rootFinder, langs, urlCommunicator, tmcApi, settings);
+        rootFinder.setReturnValue(testPath.toString());
+        URI pastePath = URI.create("https://tmc.mooc.fi/staging/paste/ynpw7_mZZGk3a9PPrMWOOQ");
```

Comment:
Replacing the URLs used would be out of scope for this issue. Currently, the URLs are defined in the various .json files in resources/json, and the same URLs are also used in other test classes. To replace all of the URLs pointing to to the production server with http://example.com should be a separate issue, I think.